sklearn/impute/tests/test_impute.py

```diff
@@ -259,7 +259,7 @@ def test_imputation_median_special_cases():
 @pytest.mark.parametrize("dtype", [None, object, str])
 def test_imputation_mean_median_error_invalid_type(strategy, dtype):
     X = np.array([["a", "b", 3], [4, "e", 6], ["g", "h", 9]], dtype=dtype)
-    msg = "non-numeric data:\ncould not convert string to float: '"
+    msg = "non-numeric data:\ncould not convert string to float:"
```

Comment:
I remember here why I removed the quote here. We reraise an error from numpy ... we could also rewrite the error message if we think this is really important.

https://github.com/scikit-learn/scikit-learn/blob/83760ab447c63be7d499e3a5bee72abf46ab0d22/sklearn/impute/_base.py#L327

```
In [1]: import numpy as np

In [2]: np.array(['a']).astype(np.float64)
--------------------------------------------------------------------------
ValueError                               Traceback (most recent call last)
Cell In[2], line 1
----> 1 np.array(['a']).astype(np.float64)

ValueError: could not convert string to float: np.str_('a')
```